src/main/java/ca/corefacility/bioinformatics/irida/ria/web/analysis/AnalysisController.java

```diff
@@ -158,9 +168,47 @@ public String getDetailsPage(@PathVariable Long submissionId, Model model, final
 		model.addAttribute("analysisType", analysisType);
 		model.addAttribute("mailConfigured", emailController.isMailConfigured());
 
+		if(submission.getAnalysisState()==AnalysisState.ERROR) {
+			model.addAttribute("previousState", getPreviousStateBeforeError(submissionId));
+		}
+
 		return "analysis";
 	}
 
+	/**
+	 * Gets the state of analysis prior to error
+	 *
+	 * @param submissionId {@link Long} identifier for an {@link AnalysisSubmission}
+	 * @return {@link String} State of analysis prior to error
+	 */
+	private AnalysisState getPreviousStateBeforeError(Long submissionId) {
+		AuditReader audit =AuditReaderFactory.get(entityManagerFactory.createEntityManager());
+		AnalysisSubmission previousRevision = null;
+
+		if(audit != null) {
+			// Get revisions from the analysis submission audit table for the submission
+			List auditResultSet = audit.createQuery()
+					.forRevisionsOfEntity(AnalysisSubmission.class, true, false)
+					.add(AuditEntity.id()
+							.eq(submissionId)).getResultList();
+
+			// Go through the revisions and find the first one with an error. The revision
+			// prior is set to the previousRevision and it's state is set in the model
+			for (Object auditResult : auditResultSet) {
+				AnalysisSubmission auditedSubmission = (AnalysisSubmission) auditResult;
+				if (auditedSubmission.getAnalysisState() == AnalysisState.ERROR) {
+					break;
+				}
+				previousRevision = auditedSubmission;
+			}
+		}
+		if(previousRevision==null) {
```

Comment:
Formatting, why do you hate spaces so much?

Reply:
Fixed in the previous commit. Forgot to run the intellij formatter on the method :)